contracts/Karmic.sol

```diff
@@ -32,16 +69,33 @@ contract Karmic is Badger{
         }
     }
 
+    function withdraw(address token, uint256 amount) external{
+        require(!boxTokenTiers[token].passedThreshold,
+            "Can withdraw only funds for tokens that didn't pass threshold");
+        uint256 withdrawnFunds = amount * boxTokenTiers[token].funds / boxTokenTiers[token].amount;
+        boxTokenTiers[token].amount -= amount;
```

Comment:
What speaks against this 👆🏼  approach, @BritikovKI ? To me it seems more efficient than the current one, because then we hardly need to save/update any state variables. Rn, we're updating the `boxTokenTiers[token].amount` for any `withdraw` or `claimGovernanceTokens`, which we could avoid, imo.

Reply:
What speaks against this ☝🏼 approach, @BritikovKI ? My understanding is that by using it we could save on two `WRITE` operations (`boxTokenTiers[token].funds` and `boxTokenTiers[token].amount`) per each `withdraw`, and one `WRITE` per each `claimGovernanceTokens` (`boxTokenTiers[token].amount`), which seems desirable to me. Or am I missing something?

Reply:
There is one thing, in this case we are supposing that ration of tokens/ether is the same for any of the boxes(1:1, in which I'm not sure), which is assured in this case, but inot in the general approach. This way of storing the value makes this contract more general in usage, because we don't care about croudfund token/ether ratio

Reply:
This thing is implemented the way we discussed, no state changes